metadata/metadata_service/proxy/neo4j_proxy.py

```diff
@@ -551,14 +562,16 @@ def _safe_get(self, dct, *keys):
     def _execute_cypher_query(self, *,
                               statement: str,
                               param_dict: Dict[str, Any]) -> List[Record]:
+        """
+        Execute Cypher queries using managed read transactions
+        """
         if LOGGER.isEnabledFor(logging.DEBUG):
             LOGGER.debug('Executing Cypher query: {statement} with params {params}: '.format(statement=statement,
                                                                                              params=param_dict))
         start = time.time()
         try:
             with self._driver.session(database=self._database_name) as session:
-                result = session.run(query=statement, **param_dict)
-                return [record for record in result]
+                return session.read_transaction(execute_statement, statement, param_dict)
```

Comment:
yes this is a good point, I was having trouble getting the results originally so I thought it needed to be done a different way, but turns out I was just doing it in the wrong spot. just updated this so the original return way is used, but just in the transaction function instead of here